R/060_utils.R

```diff
@@ -0,0 +1,348 @@
+#' paste3
+#'
+#' An internal fucntion that removes NAs from paste.
+#' Source: https://stackoverflow.com/users/1855677/42.
+#'
+#' @param ... objects to be pasted together.
+#' @param sep separating string.
+
+paste3 <- function(..., sep = ", ") {
+  L <- list(...)
+  L <- lapply(L, function(x) {
+    x[is.na(x)] <- ""
+    x
+  })
+  ret <- gsub(
+    paste0("(^", sep, "|", sep, "$)"), "",
+    gsub(
+      paste0(sep, sep), sep,
+      do.call(paste, c(L, list(sep = sep)))
+    )
+  )
+  is.na(ret) <- ret == ""
+  ret
+}
+
+#' Supply an expression to identify which header groups are hooked.
+#'
+#' @description
+#' This function is used with the `.hook_if` or `.hook_if_rev` arguments in the `locate_groups` function.
+#' It passes an expression to `dplyr::summarise` that identifies which header groups are hooked - for example swiched from N to NNW.
+#' See the the `locate_groups` documentation for more information and an example.
+#'
+#' @param ...  expression applied to a  identifies which header groups are hooked
+#'
+#' @name get_range_dfs
+#' @export
```

Comment:
Please can all exported functions have `@examples`.

Reply:
Sure. I should have got around to this earlier.

Reply:
Have added descriptions in 2ddc99f73f070b228909c5a2f5c7d819fdb7e249, 299fab1852cb79ca9b1c3d40f46554cf94d5c860, and e4280eaee61bee8195d4ad4ce8bfd661fe39cbe4.